src/storage/src/render/upsert.rs

```diff
@@ -200,19 +212,80 @@ fn evaluate(
     Ok(Some(row_buf.clone()))
 }
 
+fn extract_kv<G: Scope>(
```

Comment:
Would you mind adding a doccomment here? I started reading the code, but I couldn't confirm that it does any one specific thing.

Reply:
Also, is the logic in the body of the operator potentially valuable to extract? Maybe no, but I was surprised to see we hadn't extracted this previously!

Reply:
I can't remember having typed this exact logic when I was doing the permutations stuff this past winter, but if it exists somewhere, I'd be happy to use it!